src/test/java/uk/gov/companieshouse/exemptions/delta/InvalidMessageRouterTest.java

```diff
@@ -38,14 +47,38 @@ void setup() {
     @Test
     void testOnSendRoutesMessageToInvalidMessageTopicIfNonRetryableExceptionThrown() {
         // given
-        ProducerRecord<String, ChsDelta> message = new ProducerRecord<>("main", "key", delta);
+        ProducerRecord<String, ChsDelta> message = new ProducerRecord<>("main", 0, "key", delta,
+                List.of(
+                        new RecordHeader(ORIGINAL_TOPIC, "main".getBytes()),
+                        new RecordHeader(ORIGINAL_PARTITION, BigInteger.ZERO.toByteArray()),
+                        new RecordHeader(ORIGINAL_OFFSET, BigInteger.ONE.toByteArray()),
+                        new RecordHeader(EXCEPTION_MESSAGE, "[invalid]".getBytes())));
+
+        ChsDelta invalidData = new ChsDelta(
+                "{ \"invalid_message\": \"payload: [ invalid ] passed for topic: main, partition: 0, offset: 1\" }",
+                0, "", false);
+        // when
+        ProducerRecord<String, ChsDelta> actual = invalidMessageRouter.onSend(message);
+
+        // then
+        verify(flags, times(0)).destroy();
+        assertThat(actual, is(equalTo(new ProducerRecord<>("invalid", "key", invalidData))));
+    }
+
+    @Test
+    void testOnSendRoutesMessageToInvalidMessageTopicIfNonRetryableExceptionThrownNoHeaders() {
```

Comment:
Should this test name and the test case above mention InvalidPayloadException instead of NonRetryableException?

Reply:
Resolved in latest commit.